versioned_docs/version-2.x/magic-button/x-ray.mdx

```diff
@@ -0,0 +1,141 @@
+---
+sidebar_position: 4
+title: X-Ray
+description:
+  Enhances your stores functionality by providing an edit button alongside each
+  content. Content creators can directly access the source for editing, thereby
+  improving content management efficiency.
+---
+
+import BrowserWindow from "@site/src/components/BrowserWindow";
+import SinceVersion from "@site/src/components/SinceVersion";
+import ContactLink from "@site/src/components/ContactLink";
+
+<SinceVersion tag="2.25" />
+
+The **X-Ray** feature enhances your stores functionality by providing an edit
+button alongside each content. Content creators can directly access the source
+for editing, thereby improving content management efficiency.
+
+## How to activate X-Ray
+
+To activate the **X-Ray** feature, click on the Magic Button and
+select the X-Ray option.
+
+<div style={{ width: 600, maxWidth: "100%" }}>
+  <BrowserWindow noPadding>
+    <img
+      src={require("./assets/enable-x-ray.gif").default}
+      width="100%"
+      height="100%"
+      style={{ objectFit: "contain", marginBottom: -10 }}
+    />
+  </BrowserWindow>
+</div>
+
+When enabled, content composing the page are outlined with a user interface
+allowing the contributor to know the source of it and providing a quick access
+to the corresponding administration interface.
+
+## X-Ray on custom types
+
+By default, the X-Ray feature is available for the main GraphQL types provided
+by Front-Commerce (Product, Categories, …). It is also possible to implement
+X-Ray for custom types.
```

Comment:
the case of existing types being extended with data coming from another source is not covered